src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -140,16 +140,22 @@ internal static FunctionSignatureWithReturnType MatchSignatureToBuiltInFunction(
 
         /// <summary>
         /// Finds all signatures for the given function name.
+        /// Search in BuiltIn uri functions first, and then in custom uri functions.
         /// </summary>
         /// <param name="functionName">The function to get the signatures for.</param>
         /// <returns>The signatures which match the supplied function name.</returns>
-        internal static FunctionSignatureWithReturnType[] GetBuiltInFunctionSignatures(string functionName)
+        internal static FunctionSignatureWithReturnType[] GetUriFunctionSignatures(string functionName)
         {
-            // Try to find the function in our built-in functions
             FunctionSignatureWithReturnType[] signatures;
-            if (!BuiltInFunctions.TryGetBuiltInFunction(functionName, out signatures))
+
+            // Try to find the function in the user custom functions
+            if (!CustomUriFunctions.TryGetCustomFunction(functionName, out signatures))

```

Comment:
The method 'AddCustomUriFunction' in 'CustomUriFunctions' class gets a 'overrideBuiltInFunction' parameter which indicates whether to override the existing built-in function with the same name of the given custom function.
As I mentioned in a comment: "It overrides the BuiltIn function because search is first on CustomFunctions and then on BuiltInFunctions. Another option is to remove the BuiltIn function - seems like a worse option.".
I gave the user an option to to override the built in functions with the same name, so the OData protocol stay with no changes. I could have removed the Built-In function, but it seems like the worse option because if the user will removed a custom function, which had been overrided and removed a built-in function, there will be no "true roll-back". I define a "true roll-back" as to add the removed built-in function again.
When searching first on CustomUriFunction and then on BuiltInUriFunction, adding an overriding custom function will be easy,
so when an overriding CustomUriFunction will be removed, the built-in function will always stay the same and make sure the OData protocol standstill.

- This is why I added some tests about who is searched first.